bee-message/src/metadata.rs

```diff
@@ -0,0 +1,211 @@
+        const BOOKED = 0b0000_0100;
+        /// Denotes whether a message is eligible, i.e. it's timestamp is good.
+        const ELIGIBLE = 0b0000_1000;
+        /// Denotes whether a message has been deemed invalid, i.e. it or its parents do not pass all checks from
+        /// filters to message booker.
+        const INVALID = 0b0001_0000;
+    }
+}
+
+impl Flags {
+    /// Returns the solid flag.
+    pub fn is_solid(&self) -> bool {
+        self.contains(Flags::SOLID)
+    }
+
+    /// Sets the solid flag.
+    pub fn set_solid(&mut self, is_solid: bool) {
+        self.set(Flags::SOLID, is_solid);
+    }
+
+    /// Returns the scheduled flag.
+    pub fn is_scheduled(&self) -> bool {
+        self.contains(Flags::SCHEDULED)
+    }
+
+    /// Sets the scheduled flag.
+    pub fn set_scheduled(&mut self, is_scheduled: bool) {
+        self.set(Flags::SCHEDULED, is_scheduled);
+    }
+
+    /// Returns the booked flag.
+    pub fn is_booked(&self) -> bool {
+        self.contains(Flags::BOOKED)
+    }
+
+    /// Sets the booked flag.
+    pub fn set_booked(&mut self, is_booked: bool) {
+        self.set(Flags::BOOKED, is_booked);
+    }
+
+    /// Returns the eligible flag.
+    pub fn is_eligible(&self) -> bool {
+        self.contains(Flags::ELIGIBLE)
+    }
+
+    /// Sets the eligible flag.
+    pub fn set_eligible(&mut self, is_eligible: bool) {
+        self.set(Flags::ELIGIBLE, is_eligible);
+    }
+
+    /// Returns the invalid flag.
+    pub fn is_invalid(&self) -> bool {
+        self.contains(Flags::INVALID)
+    }
+
+    /// Sets the invalid flag.
+    pub fn set_invalid(&mut self, is_invalid: bool) {
+        self.set(Flags::INVALID, is_invalid);
+    }
+}
+
+impl Packable for Flags {
+    type PackError = Infallible;
+    type UnpackError = Infallible;
+
+    fn packed_len(&self) -> usize {
+        self.bits().packed_len()
+    }
+
+    fn pack<P: Packer>(&self, packer: &mut P) -> Result<(), PackError<Self::PackError, P::Error>> {
+        self.bits().pack(packer)
+    }
+
+    fn unpack<U: Unpacker>(unpacker: &mut U) -> Result<Self, UnpackError<Self::UnpackError, U::Error>> {
+        Ok(Self::from_bits_truncate(u8::unpack(unpacker).infallible()?))
+    }
+}
+
+/// Additional data that describes the local perception of a message which is not part of the Tangle.
+#[derive(Clone, Default, Debug, Eq, PartialEq)]
+#[cfg_attr(feature = "serde1", derive(serde::Serialize, serde::Deserialize))]
+pub struct MessageMetadata {
+    /// The flags describing the message state.
+    flags: Flags,
+    /// The local time the message was received by the node.
+    received_timestamp: u64,
+    /// The local time the message got solid.
+    solidification_timestamp: u64,
+    /// The branch ID of the message, i.e. the part of the Tangle where the message is located.
+    branch_id: [u8; 32],
+    /// Contains the node's opinion on the issuing timestamp of a message.
```

Comment:
Is Fpc really voting on the issuing timestamp of a message?

Reply:
Yes

Reply:
https://github.com/iotaledger/IOTA-2.0-Research-Specifications/blob/main/2.2%20Message%20Layout.md#226-metadata

Reply:
okey interesting, thx!